### PR TITLE
Do not require ocp_version when running deprecated_api role in a standalone mode

### PR DESCRIPTION
**Pull request by tkrishtop**
TestDallasWorkload: preflight-green

### CODE REVIEW

**Comment by tkrishtop** [comment]
Tested locally by manually copying the role:
```
├── deprecated_api.yml
├── inventory
└── roles
    └── deprecated_api
        ├── filter_plugins
        │   └── ocp_compatibility.py
        └── tasks
            ├── ensure_ocp_version.yml
            ├── get_api_request_counts_per_namespace.yml
            └── main.yml
```

and using new invocation not requiring to set OCP version anymore

```
---
- name: Connect to cluster and list nodes
  hosts: local
  tasks:
    - name: Detect to-be-Removed APIs in all namespaces excluding ones starting with openshift and kube-
      include_role:
        name: deprecated_api
      vars:
        deprecated_api_logs:
          path: "{{ playbook_dir }}"
      register: compatibility_map
```

Everything works fine:
```
$ ansible-playbook -i inventory -v deprecated_api.yml

-- snip --

TASK [deprecated_api : Set OCP version in major.minor format] *******************************************************************************************
ok: [127.0.0.1] => {"ansible_facts": {"da_ocp_version": "4.11"}, "changed": false}

-- snip --

TASK [deprecated_api : Extract deprecated and to-be-deprecated API for default] *************************************************************************
ok: [127.0.0.1] => {"ansible_facts": {"da_removed_api": [{"name": "events.v1beta1.events.k8s.io", "removedInRelease": "1.25"}, {"name": "flowschemas.v1beta1.flowcontrol.apiserver.k8s.io", "removedInRelease": "1.26"}, {"name": "podsecuritypolicies.v1beta1.policy", "removedInRelease": "1.25"}, {"name": "prioritylevelconfigurations.v1beta1.flowcontrol.apiserver.k8s.io", "removedInRelease": "1.26"}]}, "changed": false}

TASK [deprecated_api : Compute OCP compatibility of the workload API for default] ***********************************************************************
ok: [127.0.0.1] => {"ansible_facts": {"ocp_compatibility": {"4.11": "compatible", "4.12": "podsecuritypolicies.v1beta1.policy, events.v1beta1.events.k8s.io", "4.13": "podsecuritypolicies.v1beta1.policy, events.v1beta1.events.k8s.io, prioritylevelconfigurations.v1beta1.flowcontrol.apiserver.k8s.io, flowschemas.v1beta1.flowcontrol.apiserver.k8s.io"}}, "changed": false}

-- snip --

PLAY RECAP **********************************************************************************************************************************************
127.0.0.1                  : ok=14   changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   
```

**Comment by tkrishtop** [comment]
Local tests are ok, ready for the review. Would appreciate any input on how to fix the `CheckDallasWorkload` situation :)

**Comment by fredericlepied** [comment]
retest

**Comment by tkrishtop** [comment]
Invocation out of DCI [is working](https://www.distributed-ci.io/jobs/c6af5a91-e346-4b8e-8a66-1308c4b85c0f/tests/5eed5f89-a0f9-4d7e-a773-dbab198c30f8) as well.
The change is now 100% tested and ready for review.

File: roles/deprecated_api/tasks/main.yml [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
                       | map(attribute='metadata.name')
                       | select('match', '^(?!openshift|kube-).*')
                       | list }}"
 
-- name: Loop over deprecated_namespaces to get API request counts
+- name: "Loop over deprecated_namespaces to get API request counts"
   ansible.builtin.include_tasks: get_api_request_counts_per_namespace.yml
   # Check all namespaces by default unless otherwise requested
-  loop: "{{ deprecated_namespaces | default(all_namespaces) }}"
+  loop: "{{ deprecated_api_namespaces | default(da_all_namespaces) }}"
```
Comment (ramperher) [inline roles/deprecated_api/tasks/main.yml:20]: Do we have any user, partner, team, etc. that is using `deprecated_namespaces` var? Just to let them now that they have to update this.

**Review by ramperher — APPROVED** [review]
LGTM, just one note regarding the change on the main var for this role.

File: roles/deprecated_api/tasks/main.yml (hunk 1 above)
Comment (tkrishtop) [reply roles/deprecated_api/tasks/main.yml:20]: nope, nobody so far :(

**Pull request merged**